integration-tests/src/test/java/org/apache/fineract/integrationtests/DateValidationTest.java

```diff
@@ -0,0 +1,237 @@
+        final Integer loanProductID = loanTransactionHelper.getLoanProductId(loanProductJSON);
+
+        loanTransactionHelper = new LoanTransactionHelper(requestSpec, errorResponseSpec);
+
+        final String loanApplicationJSON = new LoanApplicationTestBuilder().withPrincipal("1000").withLoanTermFrequency("1")
+                .withLoanTermFrequencyAsMonths().withNumberOfRepayments("1").withRepaymentEveryAfter("1")
+                .withRepaymentFrequencyTypeAsMonths().withInterestRatePerPeriod("0").withInterestTypeAsFlatBalance()
+                .withAmortizationTypeAsEqualPrincipalPayments().withInterestCalculationPeriodTypeSameAsRepaymentPeriod()
+                .withExpectedDisbursementDate(invalidDate).withSubmittedOnDate("01 March 2022").withLoanType("individual")
+                .build(clientId.toString(), loanProductID.toString(), null);
+        HashMap<String, Object> response = (HashMap) loanTransactionHelper.createLoanAccount(loanApplicationJSON, "");
+        List<HashMap<String, Object>> errors = (List) response.get("errors");
+        assertNotNull(errors);
+        HashMap<String, Object> error = errors.get(0);
+        assertNotNull(error);
+        assertEquals(
+                "The parameter `expectedDisbursementDate` is invalid based on the dateFormat: `dd MMMM uuuu` and locale: `en_GB` provided:",
+                error.get("developerMessage"));
+    }
+
+    @Test
+    public void testShouldFailWithInvalidDateTime() {
+        String requestCode = UUID.randomUUID().toString();
+        InteropTransactionRole role = InteropTransactionRole.PAYER;
+        String requestBody = buildRequestBody(requestCode, role);
+        String response = interopHelper.postTransactionRequest(requestCode, role, requestBody);
+        HashMap<String, Object> map = new Gson().fromJson(response, new TypeToken<HashMap<String, Object>>() {}.getType());
+        List<Map<String, Object>> errors = (List) map.get("errors");
+        assertNotNull(errors);
+        Map<String, Object> error = errors.get(0);
+        assertNotNull(error);
+        assertEquals("The parameter `expiration` is invalid based on the dateFormat: `dd MMMM uuuu HH:mm:ss` and locale: `en` provided:",
+                error.get("developerMessage"));
+    }
+
+    @Test
+    public void testShouldFailWithInvalidMonthDay() {
+        final Account assetAccount = this.accountHelper.createAssetAccount();
+        final Account incomeAccount = this.accountHelper.createIncomeAccount();
+        final Account expenseAccount = this.accountHelper.createExpenseAccount();
+        final Account liabilityAccount = this.accountHelper.createLiabilityAccount();
+
+        DateFormat dateFormat = new SimpleDateFormat("dd MMMM yyyy", Locale.US);
+        new SimpleDateFormat("dd MMM", Locale.US);
+        DateFormat currentDateFormat = new SimpleDateFormat("dd");
+
+        Calendar todaysDate = Calendar.getInstance();
```

Comment:
Please use LocalDate instead of Calendar...

Reply:
fixed